dali/python/nvidia/dali/external_source.py

```diff
@@ -420,23 +420,37 @@ class ExternalSource():
 
     Acceptable sources depend on the value specified for ``batch`` parameter.
 
-    If batch is set to False, source must be a callable (a function or an object with __call__ method)
-    that accepts exactly one argument (:meth:`~nvidia.dali.types.SampleInfo` instance that
-    represents the index of the requested sample).
-    If batch is set to True, the ``source`` can be either a callable, an iterable or a generator function.
-    Callable in batch mode must accept exactly one argument - either :meth:`~nvidia.dali.types.BatchInfo`
-    instance or an integer (see `batch_info`).
+    If ``batch`` is set to ``False``, the ``source`` must be:
+
+      * a callable (a function or an object with __call__ method) that accepts exactly one argument
+        (:meth:`~nvidia.dali.types.SampleInfo` instance that represents the index of the requested sample).
+
+    If ``batch`` is set to ``True``, the ``source`` can be either:
+
+      * a callable that accepts exactly one argument (either :meth:`~nvidia.dali.types.BatchInfo`
+        instance or an integer - see ``batch_info`` for details)
+      * an iterable,
+      * a generator function.
 
     Irrespective of ``batch`` value, callables should produce requested sample or batch solely based on
     the SampleInfo/BatchInfo instance or index in batch, so that they can be run in parallel in a number of workers.
-    When batch is set to True, callables performance might especially benefit from increasing
+
+    When ``batch=True``, callables performance might especially benefit from increasing
     ``prefetch_queue_depth`` so that a few next batches can be computed in parallel.
 
-    Iterator or generator will be assigned to a single worker that will iterate over it.
+    .. note::
+        Iterators and generator functions will be assigned to a single worker that will iterate over it.
```

Comment:
I wonder if we could stay with a singular form here? `Iterator or generator function`? Because if there are two different iterators and more than one worker, they won't be assigned to the same worker.

Reply:
Done